### PR TITLE
Make LStore robust to accessor issues by using a proxy.

### PR DESCRIPTION
**Pull request by gregglind**

### CODE REVIEW

**Comment by gregglind** [comment]
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy

```javascript

// throws in strict mode
var A = Object.preventExtensions({data: 1});  

var handler = { get:  function (target, name) { if (name in target) { return target[name] } else { throw new Error("oh no")}}};
var p = new Proxy(A, handler)

p.wut   // throws
```

**Comment by gregglind** [comment]
For now at least, not going to make the 'data' into a smart object.  Too 'weird'.

File: package.json [hunk 1]
```diff
@@ -37,7 +37,8 @@
     "open": "opener",
     "webpack": "webpack --module-bind traceur-loader --config webpack.single.conf.js --watch --progress",
     "test": "node_modules/karma/bin/karma start karma.conf.js && npm run thirdparty",
-    "pretest": "npm run compile && echo > test/z-ensure-coverage.js; find src -type f -name '*.js' -print0 | xargs -0 -I '{}' echo 'require(\"../{}\");' | egrep -v \"(jetpack|main.js)\" >> test/z-ensure-coverage.js",
+    "prep_cover":  "echo > test/z-ensure-coverage.js;   git ls-tree -r HEAD --name-only src | grep \"js$\" | xargs -I '{}' echo 'require(\"../{}\");' | egrep -v \"(jetpack|main.js)\" >> test/z-ensure-coverage.js",
```
Comment (gregglind) [inline package.json]: Bug #201

File: src/common/recipe-utils.js [hunk 2]
```diff
@@ -23,6 +23,7 @@
   *
   **/
 var Lstore = function (key, storage) {
+  "use strict";
```
Comment (gregglind) [inline src/common/recipe-utils.js]: changes here all fix #200

File: test/common/test-recipe-utils.js [hunk 3]
```diff
@@ -63,7 +63,7 @@ describe("recipe-utils", function () {
       expect(localStorage[u]).equal("{}");
     });
 
-    it("is sealed to prevent setting keys on it instead of data", function () {
+    it("is preventExtensions to prevent setting keys on it instead of data", function () {
```
Comment (gregglind) [inline test/common/test-recipe-utils.js]: part of #200

File: test/common/test-recipe-utils.js [hunk 4]
```diff
@@ -87,5 +87,12 @@ describe("recipe-utils", function () {
       expect(L.revive() === L).true();
       expect(L.clear() === L).true();
     })
+
+    it("accessing and setting keys on the Lstore itself will throw.", () => {
```
Comment (gregglind) [inline test/common/test-recipe-utils.js]: part of #200

File: test/z-ensure-coverage.js [hunk 5]
```diff
@@ -13,13 +13,7 @@ require("../src/common/index.js");
 require("../src/common/personinfo.js");
 require("../src/common/recipe-utils.js");
 require("../src/common/request.js");
```
Comment (gregglind) [inline test/z-ensure-coverage.js]: effect of changes to package.json #201

**Comment by rayborn** [comment]
r+, everything looks good to me, tested locally

**Pull request merged**